core/src/main/java/org/elasticsearch/transport/local/LocalTransportChannel.java

```diff
@@ -98,17 +97,22 @@ public void sendResponse(Throwable error) throws IOException {
         RemoteTransportException tx = new RemoteTransportException(targetTransport.nodeName(),
                 targetTransport.boundAddress().boundAddresses()[0], action, error);
         stream.writeThrowable(tx);
+        sendResponseData(stream.bytes().toBytes());
+        sourceTransportServiceAdapter.onResponseSent(requestId, action, error);
+    }
 
-        final byte[] data = stream.bytes().toBytes();
+    private void sendResponseData(byte[] data) {
+        sourceTransport.inFlightRequestsBreaker.addWithoutBreaking(-reservedBytes);
+        reservedBytes = 0;
```

Comment:
can we make this send once? with same autoclosable logic suggested for the netty channel?